src/neuroglancer/synapse/frontend.ts

```diff
@@ -0,0 +1,309 @@
+  selectedColor = Float32Array.of(1.0, 1.0, 0.0, 1.0);
+
+  constructor(
+      public chunkManager: ChunkManager, public pointList: SynapseAnnotationPointList,
+      public voxelSizeObject: VoxelSize, public selectedIndex: WatchableValue<number|null>) {
+    super();
+    this.buffer = new Buffer(chunkManager.gl);
+    this.registerSignalBinding(pointList.changed.add(() => {
+      // Clear selectedIndex, since the indices have changed.
+      this.selectedIndex.value = null;
+      this.redrawNeeded.dispatch();
+    }));
+    this.registerSignalBinding(selectedIndex.changed.add(() => { this.redrawNeeded.dispatch(); }));
+  }
+
+  get gl() { return this.chunkManager.gl; }
+
+  updateBuffer() {
+    let {pointList} = this;
+    const newGeneration = pointList.generation;
+    if (this.generation !== newGeneration) {
+      this.generation = newGeneration;
+      this.buffer.setData(pointList.points.view);
+    }
+  }
+
+  updateMouseState(mouseState: MouseSelectionState, pickedOffset: number) {
+    vec3.multiply(mouseState.position, this.pointList.get(pickedOffset), this.voxelSizeObject.size);
+  }
+}
+
+export class RenderHelper extends RefCounted {
+  private shaders = new Map<ShaderModule, ShaderProgram>();
+  private squareCornersBuffer = getSquareCornersBuffer(this.gl);
+  private countingBuffer = this.registerDisposer(getCountingBuffer(this.gl));
+  private lineShader : ShaderProgram;
+
+  constructor(public gl: GL) { super(); }
+
+  defineShader(builder: ShaderBuilder) {
+    // Position of point in camera coordinates.
+    builder.addAttribute('highp vec3', 'aVertexPosition');
+
+    // XY corners of square ranging from [-1, -1] to [1, 1].
+    builder.addAttribute('highp vec2', 'aCornerOffset');
+
+    // The x and y radii of the point in normalized device coordinates.
+    builder.addUniform('highp vec2', 'uPointRadii');
+
+    builder.addUniform('highp vec4', 'uColorPre');
+    builder.addUniform('highp vec4', 'uColorPost');
+    builder.addUniform('highp vec4', 'uColorSelected');
+    builder.addUniform('highp vec4', 'uSelectedIndex');
+    builder.addVarying('highp vec4', 'vColor');
+
+    // Transform from camera to clip coordinates.
+    builder.addUniform('highp mat4', 'uProjection');
+    builder.addUniform('highp vec4', 'uPickID');
+    builder.addVarying('highp vec4', 'vPickID');
+    builder.addVarying('highp vec2', 'vPointCoord');
+    builder.addVertexCode(glsl_floatToUint32);
+    builder.require(countingBufferShaderModule);
+    builder.addVertexCode(glsl_addUint32);
+    builder.addVertexCode(glsl_divmodUint32);
+    builder.setVertexMain(`
+gl_Position = uProjection * vec4(aVertexPosition, 1.0);
+gl_Position.xy += aCornerOffset * uPointRadii * gl_Position.w;
+vPointCoord = aCornerOffset;
+
+uint32_t primitiveIndex = getPrimitiveIndex();
+
+uint32_t pickID; pickID.value = uPickID;
+
+uint32_t quotient;
+divmod(primitiveIndex, 2.0, quotient);
+vPickID = add(pickID, quotient).value;
+
+uint32_t _;
```

Comment:
yes, it an unused variable. Is there any convention for it in TypeScript, "_" is a Golang convention.